package.json

```diff
@@ -17,7 +17,8 @@
     "precise-commits": "precise-commits",
     "prepublishOnly": "npm run build",
     "start": "node index.js",
-    "test": "npm run lint",
+    "test": "mocha --timeout=20000 --bail --exit --recursive ./test/**/*.spec.js",
+    "test:coverage": "nyc --reporter html --reporter text mocha --timeout=20000 --bail --exit --recursive ./test/**/*.spec.js",
```

Comment:
While you're here, can you give a quick type-up of each flag you're using on these new commands?

Reply:
sure let's add some comments in the file itself

Reply:
I don't need comments in the file itself, I was hoping you can just explain _why_ you are making the changes you are proposing here.

If this is a new testing tool we'll be adding in, it's a good opportunity to explain your reasoning for the changes

Reply:
**Mocha**
`--timeout` -  This flag changes the default timeout of the mocha tests. 
 The default timeout of mocha for a test case is 2000 ms  I added this in the command to be obvious to any developer who's writing tests that the timeout is 2000 ms. [docs](https://mochajs.org/#-timeout-ms-t-ms).
 
 `--bail` - Causes Mocha to stop running tests after the first test failure it encounters. Corresponding “after each” and “after all” hooks are executed for potential cleanup. [docs](https://mochajs.org/#-bail-b).
 
 `--exit` - Forces Mocha process to quit after tests complete. [docs](https://mochajs.org/#-exit).
 
  `--recursive` - When looking for test files, recurse into subdirectories. [docs](https://mochajs.org/#-recursive).
  I added this because I split the test into more  than one sub-directory into test file so I need mocha to search inside all subdirectories and run all files.
  
  **NYC**
  NYC is a CLI tools for Istanbul  coverage reporter  tool.
  
  `--reporter` - this flag specified  the type of reporter you need to report the coverage on it, when text is a quick report that appear in the CLI after running test and HTML reporter produce a whole HTML project has all details about your test cases coverage status in /coverage directory that I ignored it into .gitignore. [docs](https://istanbul.js.org/docs/tutorials/mocha/)